lib/semian/redis.rb

```diff
@@ -0,0 +1,58 @@
+require 'semian'
+require 'semian/adapter'
+require 'redis'
+
+class Redis
+  class SemianError < Redis::BaseConnectionError
+    include ::Semian::AdapterError
+  end
+
+  ResourceOccupiedError = Class.new(SemianError)
+  CircuitOpenError = Class.new(SemianError)
+end
+
+module Semian
+  module Redis
+    include Adapter
+
+    ResourceOccupiedError = ::Redis::ResourceOccupiedError
+    CircuitOpenError = ::Redis::CircuitOpenError
+
+    # The naked methods are exposed as `raw_query` and `raw_connect` for instrumentation purpose
+    def self.included(base)
+      base.send(:alias_method, :raw_io, :io)
+      base.send(:remove_method, :io)
+
+      base.send(:alias_method, :raw_connect, :connect)
+      base.send(:remove_method, :connect)
+    end
+
+    def semian_identifier
+      @semian_identifier ||= begin
+        opts = options[:semian] || options['semian'.freeze] || {}
+        name = opts[:name] || opts['name'.freeze]
+        name ||= "#{location}/#{db}"
+        :"redis_#{name}"
+      end
+    end
+
+    def io(&block)
+      acquire_semian_resource(scope: :io) { raw_io(&block) }
```

Comment:
We should keep this consistent with MySQL unless it absolutely doesn't make sense.